CONTRIBUTING.md

```diff
@@ -41,11 +41,92 @@ There are only a few guidelines that we need contributors to follow.
 
    * Any other changes should be recompiled automatically by the VS Code prelaunch task.
      * The launch will be prevented if there are compilation errors.
-4. Once the extension is installed and reloaded, there will be an OpenShift Icon on the View Container, on the lines of snap mentioned below.
+4. Once the extension is launched in debug mode, there will be an OpenShift icon in the Sidebar:
 
 ![View Container OpenShift](images/view-container-icon.png)
 
-## Running the Integration Test Suite
+When you click on it, you should see the following view:
+
+![The OpenShift sidebar. It contains five tree views: an application explorer for viewing information about the cluster, a components section, a section that lists devfile registries, a serverless functions section, and a debug section](images/openshift-sidebar-august2023.png)
+
+## Debugging webviews
+
+The first thing to note about webviews is that breakpoints set in VS Code in the webview code won't work.
+This is because the extension code and the webview JavaScript are run in separate processes.
+Any code in a `.tsx` file or under `./src/webview/common` is webview code.
+Note that not all the code under `./src/webview` is webview code;
+some of it is code used on the extension side in order to load and interact with the webview.
+
+In order watch or inspect values in webview code,
+it's easiest to use `console.log()` statements.
+The output from these statements can be inspected by opening Chrome debug tools
+(by running the command "Developer: Toggle Developer Tools"),
+then selecting the "Console" tab.
+
+![The Chrome debug tools, opened to the "Console" tab](images/chrome-devtools-console.png)
+
+The Chrome debug tools can also be used to help with writing the layout of webviews.
+In order to inspect the layout, switch to the "Elements" view.
+Then, you will need to walk through the tree in order to locate the webview HTML.
+It should be nested in two `<iframe>` elements (in order to sandbox it from the rest of VS Code).
+It will look something like this:
+
+![VS Code with the "Create Component" webview open. On the right, the Chrome debug tools are open, showing a tree that represents the HTML for the webview. In VS Code, the "Create Component" webview is highlighted in blue to indicate it's being hovered over in the Chrome debug tools HTML tree view](images/chrome-devtools-webview.png)
+
+Then, you can modify the CSS and attributes of elements in order to help experiment with or debug the webview layout.
```

Comment:
help experiment with ...

Reply:
TBH I'm not really sure what you mean by this, but I tried to clean up the wording, so let me know what you think

Reply:
Ohh sorry I thought there was supposed to be something after 'with', but looks good to me now!

Reply:
Okay!